fava/serialisation.py

```diff
@@ -103,7 +107,17 @@ def deserialise_posting(posting):
             cost = pos.cost
     else:
         units, cost = None, None
-    return data.Posting(posting["account"], units, cost, price, None, None)
+    if total_price is not None:
+        if units.number == 0:
+            unit_price = 0
```

Comment:
This should be an Amount - I think it's better to fail here and throw the error that the division by zero will throw since this is quite the edge case.